### DISCUSSION TITLE
Run the application server after specific route

### DISCUSSION THREAD

**Discussion by Engr-Asad-Hussain** [DefectDojo/django-DefectDojo#8196] [Q&A] [answered]
I am deploying the defect-dojo using the README.md guide. The thing I am looking for is to host the application at specific route for example ```http://localhost:8080/dojo/``` instead of default hosting i.e., ```http://localhost:8080/```. How to configure defect-dojo and what changes I need to implement to make this work.

Your answers are highly appreciated. Thanks for the support.

**Comment by Engr-Asad-Hussain** [comment, marked as answer]
Later on I figured out how to host DefectDojo at a particular route.

To host DefectDojo at a specific route, such as ```http://localhost:8080/dojo/```, you can follow these steps:

1. Open the dojo/fixtures/system_settings.json file in your DefectDojo installation.
2. Locate the URL_PREFIX field in the JSON file. By default, it is set to an empty string ("").
3. Modify the URL_PREFIX value to dojo/. The updated line should look like this:
```json
"URL_PREFIX": "dojo/",
```
4. In your command line or terminal, navigate to the DefectDojo installation directory. Run the following command to load the updated system settings into the database:
```shell
./manage.py loaddata system_settings.json
```
This will ensure that the modified URL_PREFIX value is applied.
5. Additionally, you need to set the DD_LOGIN_REDIRECT_URL and DD_LOGIN_URL environmental variables to reflect the new URL. You can set these variables according to your deployment environment.

Once you have completed these steps, you should be able to access DefectDojo at the specified route, i.e., http://localhost:8080/dojo/.

**Reply by mtesauro** [reply]
Thanks for sharing how you did this. _Years_ ago I did this on a DefectDojo install but I haven't seen anyone ask about this for quite some time so I wasn't sure if the URL_PREFIX and associated DD_URL_PREFIX were still in settings.py or the code.

Glad to know it still works for those what want it.

**Reply by Engr-Asad-Hussain** [reply]
Yeah, it works for me @mtesauro. Following changes I did to configure at http://localhost:8080/dojo:
1. Clone the repository:
```console
https://github.com/DefectDojo/django-DefectDojo.git
cd django-DefectDojo/
```

2. In the first step of configuration I update the system_settings.json file - Modify the ```url_prefex``` value.
dojo/fixtures/system_settings.json
```json
"url_prefix": "dojo/",
```

3. Secondly, I pass environmental variables - ```DD_LOGIN_URL``` and ```DD_LOGIN_REDIRECT_URL``` to uwsgi, celerybeat and celeryworker services in the docker-compose.yaml file.
```yaml
...
- DD_LOGIN_URL: /dojo/login
- DD_LOGIN_REDIRECT_URL: /dojo
```
I have passed these two environmental variables to all 3 services uwsgi, celerybeat and celeryworker.

Note: This is the clone at commit 5dba53afdc87a9d287461da7fa7e15c970da0720 which is the latest push till now.

Optional: If you have already setup DefectDojo, you will need to update the system settings to the database. You can achieve this by running following command:
```sh
docker-compose exec uwsgi /bin/bash -c "python manage.py loaddata system_settings"
./dc-down.sh
./dc-build.sh
./dc-up.sh
```
